httpobs/scanner/utils.py

```diff
@@ -23,7 +25,7 @@ def valid_hostname(hostname: str):
     """
 
     # Block attempts to scan things like 'localhost'
-    if '.' not in hostname or 'localhost' in hostname:
+    if not SCANNER_ALLOW_LOCAL and '.' not in hostname or 'localhost' in hostname:
```

Comment:
Is this local correct?  I think it needs to be:

```python
if not SCANNER_ALLOW_LOCAL and ('.' not in hostname or 'localhost' in hostname)
```

Although easier to read this way:

```python
if ('.' not in hostname or 'localhost' in hostname) or not SCANNER_ALLOW_LOCAL:
```